libpromises/mod_custom.c

```diff
@@ -590,6 +591,11 @@ static inline bool CustomPromise_IsFullyResolved(const Promise *pp)
     for (size_t i = 0; i < attributes; i++)
     {
         const Constraint *attribute = SeqAt(pp->conlist, i);
+        if (IsClassesBodyConstraint(attribute->lval))
+        {
+            /* Not passed to the modules, handled on the agent side. */
+            continue;
+        }
```

Comment:
The way all of these are crammed into top level attributes is really unfortunate, but not your fault of course.

Please expand the list of attributes not sent to module in the spec here:

https://github.com/cfengine/core/blob/master/docs/custom_promise_types/modules.md#evaluation-details

Reply:
> Please expand the list of attributes not sent to module in the spec here:

https://github.com/cfengine/core/pull/4552